src/base/cub_ddl_log.c

```diff
@@ -0,0 +1,1206 @@
+{
+  if (ddl_audit_handle == NULL || ip_addr == NULL)
+    {
+      return;
+    }
+
+  snprintf (ddl_audit_handle->ip_addr, sizeof (ddl_audit_handle->ip_addr), ip_addr);
+}
+
+void
+cub_ddl_log_pid (const int pid)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  ddl_audit_handle->pid = pid;
+}
+
+void
+cub_ddl_log_br_name (const char *br_name)
+{
+  if (ddl_audit_handle == NULL || br_name == NULL)
+    {
+      return;
+    }
+  snprintf (ddl_audit_handle->br_name, BROKER_NAME_LEN, br_name);
+}
+
+void
+cub_ddl_log_br_index (const int index)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+  ddl_audit_handle->br_index = index;
+}
+
+void
+cub_ddl_log_sql_text (char *sql_text, int len)
+{
+  if (ddl_audit_handle == NULL || sql_text == NULL || len < 0)
+    {
+      return;
+    }
+
+  if (ddl_audit_handle->sql_text != NULL)
+    {
+      FREE_MEM (ddl_audit_handle->sql_text);
+    }
+
+  ddl_audit_handle->sql_text = (char *) MALLOC (len + 1);
+
+  memset (ddl_audit_handle->sql_text, 0x0, len + 1);
+
+  if (ddl_audit_handle->sql_text != NULL)
+    {
+      snprintf (ddl_audit_handle->sql_text, len + 1, sql_text);
+    }
+}
+
+
+void
+cub_ddl_log_stmt_type (int stmt_type)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  ddl_audit_handle->stmt_type = stmt_type;
+
+  if (cub_is_ddl_type (stmt_type) == TRUE && ddl_audit_handle->auto_commit_mode == FALSE)
+    {
+      is_executed_ddl = TRUE;
+    }
+}
+
+void
+cub_ddl_log_loaddb_file_type (char file_type)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  ddl_audit_handle->loaddb_file_type = file_type;
+}
+
+void
+cub_ddl_log_file_name (const char *file_name)
+{
+  if (ddl_audit_handle == NULL || file_name == NULL)
+    {
+      return;
+    }
+
+  strncpy (ddl_audit_handle->file_name, file_name, PATH_MAX);
+}
+
+void
+cub_ddl_log_file_line (int file_line)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  ddl_audit_handle->file_line_number = file_line;
+}
+
+void
+cub_ddl_log_err_msg (char *msg)
+{
+  if (ddl_audit_handle == NULL || msg == NULL)
+    {
+      return;
+    }
+
+  if (ddl_audit_handle->err_msg != NULL)
+    {
+      FREE_MEM (ddl_audit_handle->err_msg);
+    }
+
+  ALLOC_COPY (ddl_audit_handle->err_msg, msg);
+  cub_remove_char (ddl_audit_handle->err_msg, '\n');
+}
+
+void
+cub_ddl_log_err_code (int err_code)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  ddl_audit_handle->err_code = err_code;
+}
+
+void
+cub_ddl_log_start_time (struct timeval *time_val)
+{
+  if (ddl_audit_handle == NULL)
+    {
+      return;
+    }
+
+  if (time_val == NULL)
+    {
+      struct timeval time;
+      gettimeofday (&time, NULL);
+      memcpy (&ddl_audit_handle->exec_begin_time, &time, sizeof (struct timeval));
+    }
+  else
+    {
+      memcpy (&ddl_audit_handle->exec_begin_time, time_val, sizeof (struct timeval));
+    }
+
+  ddl_audit_handle->execute_start_time[0] = '\0';
```

Comment:
You called **cub_get_time_string ()** 3 times, line 381, 776, 891. In the line 776, 891 you didn't set first byte as 0 except above line. Is it required? I don't think so. If it is not required, pls remove line 380.

Reply:
Thank you, i will change code as your recommend.